alws/crud/products.py

```diff
@@ -288,12 +292,72 @@ async def prepare_repo_modify_dict(
         if task.status != BuildTaskStatus.COMPLETED:
             continue
         if task.rpm_module:
-            product_repo = product_repo_mapping.get((task.arch, False))
+            product_repo = product_repo_mapping.get(
+                (task.arch, False, task.platform.name)
+            )
+            if product_repo is None:
+                continue
             modify[product_repo.pulp_href].append(task.rpm_module.pulp_href)
 
     return modify
 
 
+async def set_platform_for_products_repos(
+        db: Session,
+        product: models.Product,
+) -> None:
+    repo_debug_dict = {
+        True: 'debug-dr',
+        False: '-dr',
+    }
+    # name of a product's repo:
+    # some-username-some-product-some-platform-x86_64-debug-dr
+    repos_per_platform = {
+        f"{platform.owner.username}-{product.name}-{platform.name}-"
+        f"{repo.type}-{repo_debug_dict[repo.debug]}": platform
+        for repo in product.repositories for platform in product.platforms
+    }
+    # we do nothing if all repos have platform
+    if all(repo.platform for repo in product.repositories):
+        return
+    for repo in product.repositories:
+        repo.platform = repos_per_platform[repo.name]
+    db.add(product)
+    try:
+        await db.commit()
+        await db.refresh(product)
+    except Exception:
+        await db.rollback()
+
+
+async def set_platform_for_build_repos(
+        db: Session,
+        build: models.Build,
+) -> None:
+    repo_debug_dict = {
+        True: 'debug-br',
+        False: '-br',
+    }
+    # name of a build's repo:
+    # some-platform-x86_64-some-build-id-debug-br
+    repos_per_platform = {
+        f"{task.platform.name}-{repo.arch}-{build.id}-"
+        f"{repo_debug_dict[repo.debug]}": task.platform
+        for repo in build.repos for task in build.tasks
+    }
+    # we do nothing if all repos have platform
+    if all(repo.platform for repo in build.repos):
+        return
+    for repo in build.repos:
+        repo.platform = repos_per_platform[repo.name]
+    db.add(build)
```

Comment:
I'm think that we can associate repositories with platforms in `create_build` too